### ISSUE TITLE
[PROPOSAL] Remove waitCluster in Integration Tests

### ISSUE THREAD

**Issue by nhtruong** [opensearch-project/opensearch-js#422]
Every integration test starts of by hitting [waitCluster](https://github.com/opensearch-project/opensearch-js/blob/66c3a99cb04efe1ae6607a4fc76ef04317fd3aec/test/utils/index.js#L40-L53) method, which waits for the cluster to be up and ready before the test is executed. This is redundant because we have already implemented [healthcheck](https://github.com/opensearch-project/opensearch-js/blob/main/.ci/opensearch/Dockerfile.opensearch#L9-L12) along with [autoheal](https://github.com/opensearch-project/opensearch-js/blob/main/.ci/opensearch/docker-compose.yml#L19-L28) to assure that the OS cluster is up and running before the integration workflow executes these tests.

Moreover, `waitCluster` is waiting for status of `green`, which is usually not applicable for local testing because:
Since we're testing the client functionality (mostly API endpoints), the OS cluster only has a single node (for simplicity and it's sufficient for development and testing). However, by default every index created requires 1 replica, which means it needs a 2nd node to deploy a backup copy to. This causes the index to have status of `yellow`, which leads to `waitCluster` to wait forever if there's any index already exists on the OS cluster. For most developers, it's normal to have a few existing indices on the local OS cluster for development. But we have to delete these indices or change their replicas_count to 0 if we want to run the integration tests locally.

_Note that having a yellow status on single-node cluster is normal. The cluster is still working as intended. You just dont have a backup copy of the indices. I would say this is a bug from the server standpoint: Indices shouldn't have number of replicate set to 1 by default if there's only 1 node in the cluster._

In other words, we should remove `waitCluster` because:
- It's redundant
- It makes it harder to run integration tests locally

**Comment by nhtruong** [comment]
Workarounds:
- Remove all existing indices before running the tests: `curl -X DELETE localhost:9200/_all`
- [Update settings](https://opensearch.org/docs/2.4/api-reference/index-apis/update-settings/#example) of existing indices and set ret `number_of_replicas` to 0.

**Comment by dblock** [comment]
> Workarounds to run integration tests locally:
> 
> * Remove all existing indices before running the tests: `curl -X DELETE localhost:9200/_all`
> * [Update settings](https://opensearch.org/docs/2.4/api-reference/index-apis/update-settings/#example) of existing indices and set `number_of_replicas` to 0.

I suggest 1) documenting this in developer guide with something one can just copy-paste, 2) actually fixing integration tests to ignore existing data so that they can run on top of whatever is in an existing instance of OpenSearch (new issue?)
